src/qt/editor/welcomewidget.cpp

```diff
@@ -444,89 +616,109 @@ void WelcomeWidget::updateDetails(const QString& filename) {
-        hasTitle ? annotations.getTitle() : utilqt::fromQString(info.completeBaseName());
-    auto description = htmlEncode(annotations.getDescription());
-    replaceInString(description, "\n", "<br/>");
-
-    if (titleStr.size() > 0) titleStr[0] = static_cast<char>(std::toupper(titleStr[0]));
-    replaceInString(titleStr, "-", " ");
-    replaceInString(titleStr, "_", " ");
+    const auto createdStr = utilqt::fromQString(info.birthTime().toString(dateformat));
+    const auto modifiedStr = utilqt::fromQString(info.lastModified().toString(dateformat));
+    const auto titleStr = [&]() {
+        auto str = (annotations && !annotations->getTitle().empty())
+                       ? annotations->getTitle()
+                       : utilqt::fromQString(info.completeBaseName());
+        if (!str.empty()) str[0] = static_cast<char>(std::toupper(str[0]));
+        replaceInString(str, "-", " ");
+        replaceInString(str, "_", " ");
+        return str;
+    }();
+
+    const auto description = [&]() {
+        if (!annotations) return std::string{};
+
+        auto str = htmlEncode(annotations->getDescription());
+        replaceInString(str, "\n", "<br/>");
+        return str;
+    }();
 
     Document doc;
     auto body = doc.append("html").append("body");
     body.append("div", titleStr,
                 {{"style", "font-size:45pt; color:#268bd2; margin-top:0em; padding-top:0em;"}});
 
     auto table = body.append("table", "", {{"style", "margin-bottom:1em;"}});
-    auto addrow = [&](std::string_view name, std::string_view value) {
+    auto addRow = [&](std::string_view name, std::string_view value) {
         if (value.empty()) return;
         auto tr = table.append("tr");
         tr.append("td", name, {{"style", "text-align:right;"}});
         tr.append("td", htmlEncode(value), {{"style", "color:#ebe5df; font-weight:500;"}});
     };
 
-    addrow("File", utilqt::fromQString(info.fileName()));
-    addrow("Modified", modifiedStr);
-    addrow("Created", createdStr);
-    addrow("Author", annotations.getAuthor());
-    addrow("Tags", annotations.getTags());
-    addrow("Categories", annotations.getCategories());
-    addrow("Description", description);
-
-    if (fileBroken) {
+    addRow("File", utilqt::fromQString(info.fileName()));
+    addRow("Path", utilqt::fromQString(info.path()));
+    addRow("Modified", modifiedStr);
+    addRow("Created", createdStr);
+    if (annotations) {
+        addRow("Author", annotations->getAuthor());
+        addRow("Tags", annotations->getTags());
+        addRow("Categories", annotations->getCategories());
+        addRow("Description", description);
+    } else {
         body.append("h3", "Workspace file could not be opened!",
                     {{"style", "color:firebrick;font-weight:600;"}});
     }
+
     auto content = body.append("span");
-    auto addImage = [&content](auto item) {
-        const int fixedImgHeight = 256;
+    const int maxImageSize = utilqt::emToPx(this, 32);
```

Comment:
Make it adjustable via global settings, default 32em.

Reply:
we don't have any qt settings. And I think is is better to make sure the default is good enough.